### PR TITLE
feat(search) - add DATETIME FieldType

### PR DESCRIPTION
**Pull request by aditya-radhakrishnan**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: entity-registry/src/main/java/com/linkedin/metadata/models/annotation/SearchableAnnotation.java [hunk 1]
```diff
@@ -45,14 +45,7 @@
...
   public enum FieldType {
-    KEYWORD,
-    TEXT,
-    TEXT_PARTIAL,
-    BROWSE_PATH,
-    URN,
-    URN_PARTIAL,
-    BOOLEAN,
-    COUNT
+    KEYWORD, TEXT, TEXT_PARTIAL, BROWSE_PATH, URN, URN_PARTIAL, BOOLEAN, COUNT, DATETIME
```
[the remark this answers is not in the document]
Comment (aditya-radhakrishnan) [reply entity-registry/src/main/java/com/linkedin/metadata/models/annotation/SearchableAnnotation.java:48]: Yes, the formatting changed when I used the LinkedIn IntelliJ Java Style formatter.

[the remark this answers is not in the document]
Comment (jjoyce0510) [reply entity-registry/src/main/java/com/linkedin/metadata/models/annotation/SearchableAnnotation.java:48]: This formatter sucks

[the remark this answers is not in the document]
Comment (jjoyce0510) [reply entity-registry/src/main/java/com/linkedin/metadata/models/annotation/SearchableAnnotation.java:48]: My proposal is that we stop using it. This type of change is absolutely ridiculous.

[the remark this answers is not in the document]
Comment (aditya-radhakrishnan) [reply entity-registry/src/main/java/com/linkedin/metadata/models/annotation/SearchableAnnotation.java]: Fixed!

**Review by rslanka — APPROVED** [review]
LGTM!

**Pull request merged**